### PR TITLE
make sure user actually exists before importing to Anilist

### PR DESCRIPTION
**Pull request by toyhammered**

### CODE REVIEW

File: app/models/list_import/anilist.rb [hunk 1]
```diff
@@ -33,6 +33,19 @@ class Anilist < ListImport
     }, presence: true
     # does not accept file uploads
     validates :input_file, absence: true
+    validate :ensure_user_exists, on: :create
+
+    def ensure_user_exists
+      return if input_text.blank?
+
+      @auth_token ||= get_auth_token
+      request = Typhoeus::Request.get(build_url(input_text))
```
Comment (NuckChorris) [inline app/models/list_import/anilist.rb:42]: You know you can use the methods you already have elsewhere here?

So you could remove the `@auth_token` line and just use your `get` method from below I think

Comment (toyhammered) [reply app/models/list_import/anilist.rb:42]: that returns the body though, and we need the code.

Comment (NuckChorris) [reply app/models/list_import/anilist.rb:42]: Ah, alright then!  I think that's fine then

Comment (toyhammered) [reply app/models/list_import/anilist.rb:42]: yeah I tried to use it, but just realized it was not worth me refactoring everything for something that is being run in the background anyway.

**Pull request merged**